pandas/core/indexes/interval.py

```diff
@@ -979,14 +979,25 @@ def delete(self, loc):
         return self._shallow_copy(new_left, new_right)
 
     def insert(self, loc, item):
-        if not isinstance(item, Interval):
-            raise ValueError('can only insert Interval objects into an '
-                             'IntervalIndex')
-        if not item.closed == self.closed:
-            raise ValueError('inserted item must be closed on the same side '
-                             'as the index')
-        new_left = self.left.insert(loc, item.left)
-        new_right = self.right.insert(loc, item.right)
+        if isinstance(item, Interval):
+            if item.closed != self.closed:
+                raise ValueError('inserted item must be closed on the same '
+                                 'side as the index')
+            left_insert = item.left
+            right_insert = item.right
+        elif is_scalar(item) and isna(item):
+            # GH 18295
+            if item is not self.left._na_value:
```

Comment:
so this needs a little work in ``Base.insert`` (which works, though the values should be inserted as ``_na_value``). and doesn't work with any datetimelikes (e.g. DatetimeIndex). can you make the fix for this, rather than here?